integration-tests-ex/cases/cluster.sh

```diff
@@ -137,17 +142,34 @@ function build_shared_dir {
 	chmod -R a+rwx $SHARED_DIR
 }
 
-# Each test must have a default docker-compose.yaml file which corresponds to using
+# Either generate the docker-compose file, or use "static" versions.
+function docker_file {
+
+	# If a template exists, generate the docker-compose.yaml file. Copy over the Common
+	# folder.
+	if [ -f "$CLUSTER_DIR/docker-compose.sh" ]; then
```

Comment:
The idea here is that we already have a directory structure like this:

```text
integration-test-ex/cases
- cluster
   - SomeTest
      - docker-compose.yaml
   - AnotherTest
      - docker-compose.yaml
```

Now, suppose we convert just `SomeTest` to use the generator. Then we'll have:

```text
integration-test-ex/cases
- cluster
   - SomeTest
      - docker-compose.sh
   - AnotherTest
      - docker-compose.yaml
```

Notice that the `.yaml` file is replaced with a `.sh` file. The logic in the script looks at this file structure to decide if there is a hand-created `docker-compose.yaml` file, or if we should generate one.

If we used an environment variable, then every developer and GHA test script would have to set that variable to match the files on disk. Since there is only one right answer (or the test will fail), we might as well infer that answer from the file structure rather than forcing a human to replicate that information.